### PR TITLE
test generating service

### PR DESCRIPTION
**Pull request by Darina007**

### CODE REVIEW

File: src/main/java/com/team4/testingsystem/repositories/ContentFilesRepository.java [hunk 1]
```diff
@@ -19,4 +22,12 @@
     @Modifying
     @Query(value = "DELETE from ContentFile cf WHERE cf.id = ?1")
     int removeById(Long id);
+
+    @Query("SELECT cf from ContentFile cf " +
+           "JOIN cf.questions q " +
+           "JOIN q.level l " +
+           "WHERE l.name = ?1 " +
+           "order by function('random') ")
+    List<ContentFile> getRandomFiles(String level, Pageable pageable);
```
Comment (AntonMaksimenkau) [inline src/main/java/com/team4/testingsystem/repositories/ContentFilesRepository.java]: why do we need a list of random files?

File: src/main/java/com/team4/testingsystem/services/impl/TestGeneratingServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+import java.util.Collection;
+
+@Service
+public class TestGeneratingServiceImpl implements TestGeneratingService {
+
+    private final QuestionRepository questionService;
+    private final ContentFilesServiceImpl contentFilesService;
+
+    private static final Pageable COUNT_QUESTIONS_GRAMMAR = Pageable.ofSize(12);
```
Comment (AntonMaksimenkau) [inline src/main/java/com/team4/testingsystem/services/impl/TestGeneratingServiceImpl.java]: lets move constants to property file

File: src/main/java/com/team4/testingsystem/repositories/ContentFilesRepository.java (hunk 1 above)
Comment (AntonMaksimenkau) [reply src/main/java/com/team4/testingsystem/repositories/ContentFilesRepository.java]: BTW, does the pagination works on CrudRepository extention with queries without its direct usage

**Review by AntonMaksimenkau — CHANGES_REQUESTED** [review]
Any tests here?

**Review by AntonMaksimenkau — APPROVED** [review]

File: src/main/java/com/team4/testingsystem/services/impl/TestGeneratingServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
```
Comment (AntonMaksimenkau) [inline src/main/java/com/team4/testingsystem/services/impl/TestGeneratingServiceImpl.java]: I think, 10 questions are enough

File: src/main/java/com/team4/testingsystem/services/impl/TestGeneratingServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+public class TestGeneratingServiceImpl implements TestGeneratingService {
+
+    private final QuestionRepository questionService;
+    private final ContentFilesServiceImpl contentFilesService;
+
+    @Value("${test-generating.const.question.count}")
+    private Integer count;
+
+    @Value("${test-generating.const.one-element}")
+    private Integer oneElement;
```
Comment (AntonMaksimenkau) [inline src/main/java/com/team4/testingsystem/services/impl/TestGeneratingServiceImpl.java]: One element seems to be technical value, rather the user defines constant. No need to read it from properties.

File: src/main/java/com/team4/testingsystem/services/impl/TestGeneratingServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+    @Autowired
+    public TestGeneratingServiceImpl(QuestionRepository questionService,
+                                     ContentFilesServiceImpl contentFilesService) {
+        this.questionService = questionService;
+        this.contentFilesService = contentFilesService;
+    }
+
+    @Override
+    public Test generateTest(Test test) {
```
Comment (AntonMaksimenkau) [inline src/main/java/com/team4/testingsystem/services/impl/TestGeneratingServiceImpl.java]: Generate test method that takes test from the argument seams weird

**Review by AntonMaksimenkau — CHANGES_REQUESTED** [review]

**Review by AntonMaksimenkau — APPROVED** [review]

**Review by maxim12321 — APPROVED** [review]

**Pull request merged**